src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/AccountManagementService.java

```diff
@@ -48,4 +62,57 @@ public Boolean isUserAuthorised(String userId, ListType listType, Sensitivity se
         }
         return false;
     }
+
+    public Map<String, Optional<String>> getMappedEmails(List<String> listOfUsers) {
+        try {
+            return webClient.post().uri(url + "/" + GET_USERS_EMAIL)
+                .attributes(clientRegistrationId(ACCOUNT_MANAGEMENT_API))
+                .body(BodyInserters.fromValue(listOfUsers))
+                .retrieve()
+                .bodyToMono(new ParameterizedTypeReference<Map<String, Optional<String>>>() {})
+                .block();
+        } catch (WebClientException ex) {
+            log.error(String.format("Request with body failed. With error message: %s", ex.getMessage()));
+            return Collections.emptyMap();
+        }
+    }
+
+    public String getUserInfo(String provenanceUserId) {
+        try {
+            return webClient.get().uri(url + "/account/azure/" + provenanceUserId)
+                .attributes(clientRegistrationId(ACCOUNT_MANAGEMENT_API))
+                .retrieve().bodyToMono(String.class).block();
+        } catch (WebClientException ex) {
+            log.error(String.format("Request to account management failed with error message: %s", ex.getMessage()));
+            return "Failed to find user info for user: " + provenanceUserId;
+        }
+    }
+
+    public List<String> getAllAccounts(String provenances, String role)
+        throws JsonProcessingException {
+        try {
+            String result = webClient.get().uri(String.format(
+                    "%s/account/all?provenances=%s&roles=%s", url, provenances, role))
+                .attributes(clientRegistrationId(ACCOUNT_MANAGEMENT_API))
+                .retrieve().bodyToMono(String.class).block();
+            return findAllSystemAdmins(result);
+        } catch (WebClientException ex) {
+            log.error(String.format("Request to account management failed with error message: %s", ex.getMessage()));
+            return List.of("Failed to find all the accounts");
+        }
+    }
+
+    private List<String> findAllSystemAdmins(String result) throws JsonProcessingException {
```

Comment:
I think we should use a model for this here to capture the account return (Happy this being copied into this repo for now with the aim to clean it up later in the overall model refactoring ticket)

Reply:
fixed, now it is using PiUser model.